src/main/java/com/basiscomponents/db/exportconfig/SheetConfiguration.java

```diff
@@ -0,0 +1,186 @@
+	 *                          customizations
+	 * @param index:            position of the new column with 0-based indexing
+	 * @throws Exception in case any parameter found invalid
+	 */
+	public void insertColumn(String header, ColumnConfigurationBuilder colConfigBuilder, int index) throws Exception {
+		insertNewColumn(header, colConfigBuilder, index);
+	}
+
+	/**
+	 * Removes the column at the given position
+	 * 
+	 * @param index: position of the new column with 0-based indexing
+	 * @throws Exception in case any parameter found invalid
+	 */
+	public void removeColumn(int index) throws Exception {
+		validateIndex(index);
+
+		columnConfigs.remove(index);
+	}
+
+	/**
+	 * Removes all columns
+	 */
+	public void clearColumns() {
+		columnConfigs.clear();
+	}
+
+	/**
+	 * Retrieves the column at the given position
+	 * 
+	 * @param index: position of the new column with 0-based indexing
+	 * @throws Exception in case any parameter found invalid
+	 * @return ColumnConfiguration
+	 */
+	public ColumnConfiguration getColumn(int index) throws Exception {
+		validateIndex(index);
+
+		return columnConfigs.get(index);
+	}
+
+	/**
+	 * Gets the list of column names maintaining intended order
+	 * 
+	 * @return List of String containing column names
+	 */
+	public List<String> getColumnNamesOrdered() {
+		List<String> colNames = new ArrayList<String>();
+		for (ColumnConfiguration colConfig : columnConfigs) {
+			colNames.add(colConfig.getHeader());
+		}
+		return colNames;
+	}
+
+	/**
+	 * Applies configurations to sheet
+	 * 
+	 * @param sheet: provided Sheet object to configure
+	 * @return configured Sheet object
+	 */
+	public Sheet getConfiguredSheet(Sheet sheet) {
+		for (int index = 0; index < columnConfigs.size(); index++) {
+			int width = columnConfigs.get(index).getWidth();
+			if (width < 0)
+				continue; // Keeping default column width in case of negative width
+
+			sheet.setColumnWidth(index, width);
+		}
+
+		return sheet;
+	}
+
+	/**
+	 * Creates a new column and inserts it into given position abstracted away from
+	 * exposed methods
+	 * 
+	 * @param header:           label of the new column to be added
+	 * @param colConfigBuilder: ColumnConfigurationBuilder object containing column
+	 *                          customizations
+	 * @param index:            position of the new column with 0-based indexing
+	 * @throws Exception in case any parameter found invalid
+	 */
+	private void insertNewColumn(String header, ColumnConfigurationBuilder colConfigBuilder, int index)
+			throws Exception {
+
+		validateIndex(index);
+		ColumnConfiguration colConfig = colConfigBuilder.setHeader(header).build();
+
+		columnConfigs.add(index, colConfig);
+	}
+
+	/**
+	 * Checks if the index provided as parameter is valid
+	 * 
+	 * @param index: provided index
+	 * @throws IndexOutOfBoundsException
+	 */
+	private void validateIndex(int index) throws IndexOutOfBoundsException {
```

Comment:
So, should I rather throw the IndexOutOfBoundsException from the method calling this validIndex()?

Reply:
As i saw you call this method before you make changes to the list, which by itself throws the `IndexOutOfBoundsException` if it isn't a valid index. So you would only need this method to check before you try to change something and handle this situation. But as it is right now I can't see the benefit of this method.

Because it is a RuntimeException you also don't have to declare it in the Method signature of the calling Methods.

Reply:
Understood